Fixes #136 Use position to determine activeparam for signature help

ramya-rao-a opened the pull request:
cc @keegancsmith 

Fixes #136

ramya-rao-a commented on langserver/signature.go:
```diff
@@ -37,6 +37,28 @@ func (h *LangHandler) handleTextDocumentSignatureHelp(ctx context.Context, conn
 		activeParameter = activeParameter - 1
 	}
+
+	filename := h.FilePath(params.TextDocument.URI)
```
All this is already done in `typecheck()`. 
We can just return the `start` value from `typecheck()`
But before I went about doing that I wanted to know if there is any simpler way of getting the `Pos` equivalent of given `lsp.Position` ?

ramya-rao-a commented on langserver/signature.go:
```diff
@@ -37,6 +37,28 @@ func (h *LangHandler) handleTextDocumentSignatureHelp(ctx context.Context, conn
+	if start == token.NoPos {
+		return nil, nil
+	}
+
```
All of the above is already done in `typecheck()`. 
We can just return the `start` value from `typecheck()`
But before I went about doing that I wanted to know if there is any simpler way of getting the `Pos` equivalent of given `lsp.Position` ?

ramya-rao-a commented:
Am not sure why the test is failing, @keegancsmith can you help?

keegancsmith commented:
The test that is failing is related to the signature code.
```
package p; func main() { B(); A(); A(0,) }
                                       ^
```
The old code would say you are asking about the 2nd parameter. After your change we say you are asking about the first parameter. Hence the failing test case.

ramya-rao-a commented:
@keegancsmith Turns out the test case was using wrong position. It was using 1-based line number but 0 based character/column number. The tests passed due to the bug I am trying to fix :)

Updated the column number to be 1-based and added more test cases.

ramya-rao-a commented:
@keegancsmith `posForFileOffset` needs `offset` which you can get from `offsetForPosition` which needs file contents for which I would have to do a `readFile`
All that is already done in `typecheck`, so why do all that one more time?

Rebase is done
